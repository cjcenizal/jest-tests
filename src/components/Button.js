import React from 'react';

export const Button = ({ children, disabled, onClick, ...rest }) => {
  const onClickButton = e => {
    if (disabled) {
      return;
    }

    onClick(e);
  }

  return (
    <button disabled={disabled} onClick={onClickButton} {...rest}>
      {children}
    </button>
  );
};
