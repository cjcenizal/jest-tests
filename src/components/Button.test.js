import React from 'react';
import { render } from 'enzyme';
import { Button } from './Button';

describe('Button', () => {
  test('is rendered', () => {
    const component = render(<Button>Hello world</Button>);

    expect(component).toMatchSnapshot();
  });
});
