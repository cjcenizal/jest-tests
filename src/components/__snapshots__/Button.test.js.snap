// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is rendered 1`] = `
<button>
  Hello world
</button>
`;
